pkg/systemd/loginMessage.jsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import * as timeformat from "timeformat";
+
+const _ = cockpit.gettext;
+
+export class LoginMessage extends React.PureComponent {
```

Comment:
Using `React.PureComponent` is a new style in cockpit. We usually use functional style, which would seem fine here. You can keep the two helpers as subfunctions inside of it.